editor/components/rich-text/index.js

```diff
@@ -404,7 +359,14 @@ export class RichText extends Component {
 	 */
 
 	onChange() {
-		this.savedContent = this.getContent();
+		const record = this.getContent();
+
+		this.savedContent = this.patterns.reduce( ( accu, transform ) => transform( accu ), record );
```

Comment:
Does this mean that transforms due to matched patterns will be persisted in the post content? That seems clear here but is different than I've been mentally modeling this.

Here's my understanding of how we would apply this for @-mentions:
* **Adding formatting**: Text matching a pattern like `/@\w+/` would be formatted as an @-mention by wrapping it like `<span class="user-mention">`.
* **Removing formatting**: A `<span class="user-mention">` whose text content does not begin with an `@` would have the formatting removed by replacing the span with its text content.

Is that roughly correct?

Reply:
This one currently is just for the patterns without the mentions in mind. They're a bit different. The patterns expect a piece of text to be transformed from one thing to another. The mentions merely expect some piece of text to be displayed differently (and maybe allow to have some meta info). I'll look at the mentions shortly.

Reply:
Cool. Thanks for clarifying.